sql-parser/src/test/java/io/crate/sql/parser/TestStatementBuilder.java

```diff
@@ -749,6 +749,21 @@ public void testArrayConstructorStmtBuilder() {
         printStatement("select ARRAY[col_a IS NULL, col_b IS NOT NULL]");
     }
 
+    @Test
+    public void testArrayConstructorSubSelectBuilder() throws Exception {
+        printStatement("select array(select foo from f1) from f2");
+        printStatement("select array(select * from f1) as array1 from f2");
+        printStatement("select count(*) from f1 where f1.array1 = array(select foo from f2)");
+    }
+
+
+    @Test
+    public void testArrayConstructorSubSelectBuilderNoSubQueryThrowsParsingException() throws Exception {
+        expectedException.expect(ParsingException.class);
```

Comment:
afaik this will only test the first statement and then exit, not verifying the second case